form/src/main/java/org/dhis2/form/data/GeometryController.kt

```diff
@@ -0,0 +1,62 @@
+package org.dhis2.form.data
+
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import org.dhis2.form.model.FieldUiModel
+import org.dhis2.form.model.RowAction
+import org.hisp.dhis.android.core.arch.helpers.GeometryHelper
+import org.hisp.dhis.android.core.common.FeatureType
+import org.hisp.dhis.android.core.common.Geometry
+
+class GeometryController {
+
+    fun generateLocationFromCoordinates(featureType: FeatureType, coordinates: String?): Geometry? {
```

Comment:
I think unit test is missing. Perhaps you could pass Gson as default parameter in the constructor? 

Do we want to serialize nulls? https://futurestud.io/tutorials/gson-builder-force-serialization-of-null-values